py-polars/tests/unit/operations/namespaces/temporal/test_datetime.py

```diff
@@ -1427,3 +1427,42 @@ def test_literal_from_datetime(
 
     assert out.schema == OrderedDict({"literal": dtype})
     assert out.item() == value
+
+
+@pytest.mark.parametrize(
+    "value",
+    [
+        time(0),
+        time(hour=1),
+        time(hour=16, minute=43, microsecond=500),
+        time(hour=23, minute=59, second=59, microsecond=999999),
+    ],
+)
+def test_literal_from_time(value: time) -> None:
```

Comment:
These tests can be better. You should call `explain(optimized=False)` and check there is no `cast` in there.

Reply:
@ritchie46 can you explain a bit more? Even on the current main branch I don't see a cast in there:

```pycon
>>> import polars as pl
>>> from datetime import time
>>> print(pl.LazyFrame().select(pl.lit(time(0))).explain(optimized=False))
 SELECT [00:00:00] FROM
  DF []; PROJECT */0 COLUMNS; SELECTION: None
```

In fact, I'm having trouble coming up with *any* scenario where a CAST shows up in `explain=False` when creating a lit, unless I do something like going through categoricals:

```pycon
>>> print(pl.LazyFrame().select(pl.lit("1", dtype=pl.String).cast(pl.UInt8)).explain(optimized=False))
 SELECT [1] FROM
  DF []; PROJECT */0 COLUMNS; SELECTION: None
```